### ISSUE TITLE
How to use this code?

### ISSUE THREAD

**Issue by AIROBOTAI** [fengfu-chris/caffe-memory-optimization#2]
Hi Chris,

I accidentally come across your codes for optimizing caffe memory. Does it support training on multiple GPUs? Can you show me how to use it?  Thanks!

**Comment by fengfu-chris** [comment]
Hi @AIROBOTAI 

It suppots multiple GPUs training. The idea is to reset the upper layer memories during backpropagation and reallocate them before forwardpropagation. Besides, the batchnorm layers need to much memories for data transformation, and can be optimized to only use half of the original memories.
 
Please search MEMOPT, it will give you some clue for using this code.
